mease_elabftw/nwb.py

```diff
@@ -66,11 +73,39 @@ def get_nwb_metadata(experiment_id):
             f = json.loads(item.get("metadata", "{}")).get("extra_fields")
             for key, value in f.items():
                 print(key, value["value"])
-                # date of birth needs to be converted to datetime
+                # Date of birth needs to be converted to datetime.
                 if key.split(".")[1] == "date_of_birth":
                     metadata["Subject"][key.split(".")[1]] = datetime.fromisoformat(
                         value["value"]
                     )
+                # Mouse weight must always be given in g and is automatically converted to kg for pynwb.
+                elif key.split(".")[1] == "weight":
+                    weight_str = value["value"].lower()
```

Comment:
I would refactor this block of code into a `parse_weight(str)` or similar function,
with its own tests to cover the various possible input strings